core/src/main/java/io/questdb/cutlass/http/processors/PrometheusMetricsProcessor.java

```diff
@@ -24,42 +24,150 @@
-        HttpChunkedResponseSocket r = context.getChunkedResponseSocket();
+        final RequestState state = setupState(context);
+
+        // We double-buffer the metrics response.
+        // This is because we send back chunked responses and each chunk may not be large enough.
+        metrics.scrapeIntoPrometheus(state.sink);
+
+        final HttpChunkedResponseSocket r = context.getChunkedResponseSocket();
         r.status(200, CONTENT_TYPE_TEXT);
         r.sendHeader();
+        sendResponse(r, state);
+    }
 
-        metrics.scrapeIntoPrometheus(r);
-
-        r.done();
+    @Override
+    public boolean requiresAuthentication() {
+        return requiresAuthentication;
     }
 
+    /**
+     * Continues after `PeerIsSlowToReadException` was thrown
+     * by `onRequestComplete` or earlier call to `resumeSend`.
+     */
     @Override
     public void resumeSend(HttpConnectionContext context) throws PeerDisconnectedException, PeerIsSlowToReadException {
+        // Send the remainder of the current, partially sent, chunk.
         context.resumeResponseSend();
+
+        // Send any remaining chunks, if any.
+        final RequestState state = LV.get(context);
+        assert state != null;
+        final HttpChunkedResponseSocket r = context.getChunkedResponseSocket();
+        sendResponse(r, state);
     }
 
-    @Override
-    public boolean requiresAuthentication() {
-        return requiresAuthentication;
+    /** Send the next chunk and return true if no more chunks need to be sent. */
+    private boolean sendNextChunk(HttpChunkedResponseSocket r, RequestState state) throws PeerIsSlowToReadException, PeerDisconnectedException {
+        final int pending = state.sink.length() - state.written;
+        final int wrote = r.writeBytes(state.sink.ptr() + state.written, pending);
+        state.written += wrote;
+        final boolean done = wrote == pending;
+        r.sendChunk(done);  // Will raise `PeerIsSlowToReadException` if the tcp send buffer is full.
+        return done;
+    }
+
+    private void sendResponse(HttpChunkedResponseSocket r, RequestState state) throws PeerIsSlowToReadException, PeerDisconnectedException {
+        boolean done;
+        do {
+            done = sendNextChunk(r, state);
+        } while (!done);
+    }
+
+    private RequestState setupState(HttpConnectionContext context) {
+        RequestState state = pool.take();
+        LV.set(context, state);
+        return state;
+    }
+
+    /**
+     * State for processing a single request across multiple response chunks.
+     * Each object is used for the lifetime of one request, then returned the pool.
+     */
+    private static class RequestState implements QuietCloseable {
+        /**
+         * Metrics serialization destination, sent into one or more chunks later.
+         */
+        public final DirectByteCharSink sink = new DirectByteCharSink(4096);  // One page on most systems.
```

Comment:
```
Files.PAGE_SIZE
```

Reply:
Done. Didn't know that was there. Thanks!